File: scratch/sharpurs-adt-multi-native-20260909/after/Data.String.Common.fs
[<AutoOpen>]
module PureScript_Data_String_Common

open System
open System.Collections.Generic

let Data_String_Common__localeCompare = box (fun (arg0: obj) -> box (fun (arg1: obj) -> box (fun (arg2: obj) -> box (fun (arg3: obj) -> box (fun (arg4: obj) -> box (Data.String.Common.FFI._LocaleCompare(unbox arg0, unbox arg1, unbox arg2, unbox arg3, unbox arg4)))))))
let Data_String_Common_joinWith = box (fun (arg0: obj) -> box (fun (arg1: obj) -> box (Data.String.Common.FFI.JoinWith(unbox arg0, unbox arg1))))
let Data_String_Common_replace = box (fun (arg0: obj) -> box (fun (arg1: obj) -> box (fun (arg2: obj) -> box (Data.String.Common.FFI.Replace(unbox arg0, unbox arg1, unbox arg2)))))
let Data_String_Common_replaceAll = box (fun (arg0: obj) -> box (fun (arg1: obj) -> box (fun (arg2: obj) -> box (Data.String.Common.FFI.ReplaceAll(unbox arg0, unbox arg1, unbox arg2)))))
let Data_String_Common_split = box (fun (arg0: obj) -> box (fun (arg1: obj) -> box (Data.String.Common.FFI.Split(unbox arg0, unbox arg1))))
let Data_String_Common_toLower = box (fun (arg0: obj) -> box (Data.String.Common.FFI.ToLower(unbox arg0)))
let Data_String_Common_toUpper = box (fun (arg0: obj) -> box (Data.String.Common.FFI.ToUpper(unbox arg0)))
let Data_String_Common_trim = box (fun (arg0: obj) -> box (Data.String.Common.FFI.Trim(unbox arg0)))


let Data_String_Common_null  = (box (fun (s: obj) -> (sharpurs_apply (box ((sharpurs_apply (box ((sharpurs_apply (box ((box Data_Eq_eq))) (box ((box Data_Eq_eqString)))))) (box ((box s)))))) (box ((box ""))))))

let Data_String_Common_localeCompare  = (sharpurs_apply (box ((sharpurs_apply (box ((sharpurs_apply (box ((box Data_String_Common__localeCompare))) (box ((box Data_Ordering_LTusd_Ctor)))))) (box ((box Data_Ordering_EQusd_Ctor)))))) (box ((box Data_Ordering_GTusd_Ctor))))

File: scratch/fable_test/Data.String.Common.cs
using System;

namespace Data.String.Common;

public static class FFI {
    public static object _LocaleCompare(object lt, object eq, object gt, object s1, object s2) {
        int result = string.Compare((string)s1, (string)s2, StringComparison.CurrentCulture);
        if (result < 0) return lt;
        if (result > 0) return gt;
        return eq;
    }
    public static object Replace(object arg1, object arg2, object arg3) {
        string search = (string)arg1;
        string replace = (string)arg2;
        string target = (string)arg3;
        if (search == "") return replace + target;
        int pos = target.IndexOf(search, StringComparison.Ordinal);
        if (pos < 0) return target;
        return target.Substring(0, pos) + replace + target.Substring(pos + search.Length);
    }
    public static object ReplaceAll(object arg1, object arg2, object arg3) {
        return ((string)arg3).Replace((string)arg1, (string)arg2);
    }
    public static object Split(object arg1, object arg2) {
        string sep = (string)arg1;
        string s = (string)arg2;
        if (string.IsNullOrEmpty(sep)) {
            if (string.IsNullOrEmpty(s)) return new object[0];
            var arr = new object[s.Length];
            for (int i = 0; i < s.Length; i++) {
                arr[i] = s[i].ToString();
            }
            return arr;
        } else {
            string[] parts = s.Split(new[] { sep }, StringSplitOptions.None);
            object[] res = new object[parts.Length];
            for (int i = 0; i < parts.Length; i++) {
                res[i] = parts[i];
            }
            return res;
        }
    }
    public static object ToLower(object arg1) {
        return ((string)arg1).ToLowerInvariant();
    }
    public static object ToUpper(object arg1) {
        return ((string)arg1).ToUpperInvariant();
    }
    public static object Trim(object arg1) {
        return ((string)arg1).Trim();
    }
    public static object JoinWith(object arg1, object arg2) {
        return string.Join((string)arg1, (object[])arg2);
    }
}
